desk/app/reel.hoon

```diff
@@ -221,7 +301,126 @@
+      [%update %group ~]
+    ?>  ?=(%poke-ack -.sign)
+    ?~  p.sign  `this
+    %-  (fail:log %poke-ack 'group meta update failed' u.p.sign)
+    `this
+  ::
+      [%contacts ~]
+    ?+    -.sign  (on-agent:def wire sign)
+        %kick
+      =^  caz=(list card)  subs
+        (~(subscribe s [subs bowl]) /contacts [our.bowl %contacts] contacts-path &)
+      [caz this]
+    ::
+        %watch-ack
+      ?~  p.sign  `this
+      %-  (fail:log %watch-ack 'failed to subscribe to contacts' u.p.sign)
+      `this
+    ::
+        %fact
+      =+  !<(=response:t q.cage.sign)
+      ?.  ?=(%self -.response)  `this
+      =*  profile  con.response
+      =>
+        |%
+        ::  +hand: choose right and test equality
+        ++  hand
+          |*  [a=(unit) b=(unit)]
+          ^-  [_b ?]
+          ?~  a  [b ?=(~ b)]
+          ?~  b  [~ ?=(~ a)]
+          [b =(u.a u.b)]
+        --
+      ::  check profile for relevant changes
+      ::
+      =/  [nickname=(unit @t) axe=?]
+        %+  hand
+          (~(get cy:t our-profile) %nickname %text)
+        (~(get cy:t profile) %nickname %text)
+      =/  [avatar=(unit @t) bax=?]
+        %+  hand
+          (~(get cy:t our-profile) %avatar %look)
+        (~(get cy:t profile) %avatar %look)
+      =/  [color=(unit @ux) cax=?]
+        %+  hand
+          (~(get cy:t our-profile) %color %tint)
+        (~(get cy:t profile) %color %tint)
+      =.  our-profile  profile
+      ::  nothing relevant has changed, skip the update
+      ?:  &(axe bax cax)  `this
+      =|  update=metadata:reel
+      =.  tag.update  'group-0'
+      =.  fields.update
+        %-  ~(gas by *(map cord cord))
+        :~  'inviterNickname'^(fall nickname '')
+            'inviterAvatarImage'^(fall avatar '')
+            'inviterColor'^?^(color (rsh [3 2] u.color) '')
+        ==
+      ::  update our lure links with new nickname and avatar image
+      ::
+      =.  our-metadata
+        %-  ~(run by our-metadata)
+        |=  meta=metadata:reel
+        meta(fields (~(uni by fields.meta) fields.update))
+      ::  request the bait provider to update our invite links
+      ::
+      =/  caz=(list card)
+        %+  turn  ~(tap by our-metadata)
+        |=  [=token:reel *]
+        [%pass /update/contact %agent [civ %bait] %poke bait-update+!>([token update])]
+      [caz this]
+    ==
+  ::
+      [%groups ~]
+    ?+    -.sign  (on-agent:def wire sign)
+        %kick
+      =^  caz=(list card)  subs
+        (~(subscribe s [subs bowl]) /groups [our.bowl %groups] groups-path &)
+      [caz this]
+    ::
+        %watch-ack
+      ?~  p.sign  `this
+      %-  (fail:log %watch-ack 'failed to subscribe to groups' u.p.sign)
+      `this
+    ::
+        %fact
+      =+  !<(=r-groups:v7:groups-ver q.cage.sign)
+      ?.  ?=(%meta -.r-group.r-groups)  `this
+      =*  flag  flag.r-groups
+      =*  meta  meta.r-group.r-groups
+      =+  id=(rap 3 (scot %p p.flag) '/' q.flag ~)
+      ?~  token=(~(get by stable-id) id)  `this
```

Comment:
So, we're not sending new group metadata to bait if we ourselves haven't generated an invite link for the group. Is it possible that we haven't, but other group members have? We want to make sure to send the new metadata in that case too.